delete.ps1

```diff
@@ -140,231 +139,216 @@ function Get-ErrorMessage {
             }
-
-            break
         }
+        catch {
+            $ex = $PSItem
+            $errorMessage = Get-ErrorMessage -ErrorObject $ex
+                
+            Write-Verbose "Error at Line '$($ex.InvocationInfo.ScriptLineNumber)': $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+            
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Error updating AFAS employee [$($currentAccount.Medewerker)]. Error Message: $($errorMessage.AuditErrorMessage). Old values: $($changedPropertiesObject.oldValues | ConvertTo-Json -Depth 10). New values: $($changedPropertiesObject.newValues | ConvertTo-Json -Depth 10)"
+                    IsError = $true
+                })
+        }
+
+        break
     }
+    'NoChanges' {
+        Write-Verbose "No changes needed for AFAS employee [$($currentAccount.Medewerker)]"
+
+        if (-not($dryRun -eq $true)) {
+            # Set aRef object for use in futher actions
+            $aRef = [PSCustomObject]@{
+                Medewerker     = $currentAccount.Medewerker
+                Persoonsnummer = $currentAccount.Persoonsnummer
+            }
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "No changes needed for AFAS employee [$($currentAccount.Medewerker)]"
+                    IsError = $false
+                })
+        }
+        else {
+            Write-Warning "DryRun: No changes needed for AFAS employee [$($currentAccount.Medewerker)]"
+        }
 
-    # Define ExportData with account fields and correlation property 
-    $exportData = $account.PsObject.Copy() | Select-Object $storeAccountFields
-    # Add correlation property to exportdata
-    $exportData | Add-Member -MemberType NoteProperty -Name $correlationProperty -Value $correlationValue -Force
-    # Add aRef properties to exportdata
-    foreach ($aRefProperty in $aRef.PSObject.Properties) {
-        $exportData | Add-Member -MemberType NoteProperty -Name $aRefProperty.Name -Value $aRefProperty.Value -Force
+        break
     }
 }
-finally {
-    # Check if auditLogs contains errors, if no errors are found, set success to true
-    if (-NOT($auditLogs.IsError -contains $true)) {
-        $success = $true
-    }
-    
-    # Send results
-    $result = [PSCustomObject]@{
-        Success          = $success
-        AccountReference = $aRef
-        AuditLogs        = $auditLogs
-        PreviousAccount  = $previousAccount
-        Account          = $account
-    
-        # Optionally return data for use in other systems
-        ExportData       = $exportData
-    }
-    
-    Write-Output ($result | ConvertTo-Json -Depth 10)  
-}
\ No newline at end of file
+
+# Check if auditLogs contains errors, if no errors are found, set success to true
+if (-NOT($outputContext.AuditLogs.IsError -contains $true)) {
```

Comment:
because at the start of the script $outputContext.Success = $true is set, even if the auditlogs contains IsError $true, then the $outputContext.Success is not set to $false?

Reply:
Copy error. resolved in new commit